core/main/src/broker/broker_utils.rs

```diff
@@ -72,6 +72,17 @@ impl BrokerUtils {
         }
     }
 
+    pub async fn process_for_app_main_request(
```

Comment:
Maybe you don't have the latest from main but there is already a function, [process_internal_request()](https://github.com/rdkcentral/Ripple/blob/main/core/main/src/broker/broker_utils.rs#L83) that does this.

Reply:
This method is a derivative method which has no call context so this method offers support for those instances where callcontext is not fully available
```
 async fn report_app_state_transition(
        &mut self,
        app_id: &str,
        method: &AppMethod,
        app_manager_response: Result<AppManagerResponse, AppError>,
    )
```